libusb/core.c

```diff
@@ -2225,25 +2226,24 @@ int API_EXPORTED libusb_set_option(libusb_context *ctx,
 			ctx->debug = (enum libusb_log_level)arg;
 #endif
 		break;
-
-		/* Handle all backend-specific options here */
-	case LIBUSB_OPTION_USE_USBDK:
-	case LIBUSB_OPTION_NO_DEVICE_DISCOVERY:
-	case LIBUSB_OPTION_WEAK_AUTHORITY:
+	/* Handle all backend-specific options here */
+	case LIBUSB_OPTION_ANDROID_JNIENV:
+	case LIBUSB_OPTION_ANDROID_JAVAVM:
 		if (NULL == ctx) {
+			usbi_mutex_static_lock(&default_context_lock);
+			if (usbi_backend.set_option)
+				r = usbi_backend.set_option(ctx, option, ap);
+			else
+				r = LIBUSB_ERROR_NOT_SUPPORTED;
+			usbi_mutex_static_unlock(&default_context_lock);
+			return r;
+		}
+		else {
 			return LIBUSB_SUCCESS;
 		}
```

Comment:
Had to rework this a little to get things working.

Not completely clean 
But works